src/main/java/io/supertokens/multitenancy/MultitenancyHelper.java

```diff
@@ -193,7 +195,14 @@ private void refreshCronjobs() {
 
     public TenantConfig[] getAllTenants() {
         try {
-            return main.getResourceDistributor().withResourceDistributorLockWithReturn(() -> this.tenantConfigs);
+            return main.getResourceDistributor().withResourceDistributorLockWithReturn(() -> {
+                TenantConfig[] tenantConfigs = new TenantConfig[this.tenantConfigs.length];
```

Comment:
add comment as to why this copy is happening

Reply:
Done